### DISCUSSION TITLE
How to add postgres version15 to Laragon?

### DISCUSSION THREAD

**Discussion by DesarrollosDimpulsoX** [leokhoa/laragon#386] [General]
I need to use Laragon with the latest version of postgres, till today its version 15.1.. could you help me please?

**Comment by inckmo** [comment]
**Step-1:**  Download PGSQL from [Postgres Binaries](https://www.enterprisedb.com/download-postgresql-binaries) 

**Step-2** Extract Zip File with foldername and version number e.g. => **postgresql-15.1**

**Step-3** Create a new folder in your default **bin** Location of Laragon  =>  "**C:\laragon\bin**"  folder with Name "**postgresql**" .

**Step-4** Move Extract Zip folder to "**postgresql**" folder inside, then start your Laragon you can see postgresql there.


Default Username: postgres
Default Password: 
Default Host: localhost
Default Port: 5432

**Comment by Pen-y-Fan** [comment]
Laragon can download and install version 15,1 by adding the following **postgresql-15 ...** line under the existing **postgresql-14 ....** in laragon\usr\packages.conf (Laragon menu > Tools > Quick Add > configuration)

```ini
# PostgreSQL
postgresql-14=https://get.enterprisedb.com/postgresql/postgresql-14.5-1-windows-x64-binaries.zip
postgresql-15=https://get.enterprisedb.com/postgresql/postgresql-15.1-1-windows-x64-binaries.zip
```

You can download and install using Laragon menu > Tools > Quick Add > postgresql 15.

![postgresql ](https://user-images.githubusercontent.com/40126936/216164642-7cc95083-b433-4ee4-b245-15966656e3fd.png)

Start PostgreSQL (Laragon > PostgreSQL > Start) Wait a few seconds for it to be configured.

To access using HeidiSQL 

- click the **Database** button
- click New
- name it **Laragon.PostgreSQL**.  
- Network type is **PostreSQL (TCP/IP)** 
- library **libpq-10.dll**. 
- Save
- Open

![HeidiSQL](https://user-images.githubusercontent.com/40126936/216165397-7fd55d0f-6ba2-4732-8cc2-03cc9c8da944.png)

The client will open and you can access Heidi PostreSQL client

![HeidiSQL](https://user-images.githubusercontent.com/40126936/216166878-d680658a-8b20-4dfb-aaa7-a30d1738e3c3.png)

For details on how to use HeidiSQL click Help > [General Help](https://www.heidisql.com/help.php?place=menuReadme).

**Reply by mkanas** [reply]
i got FATAL: role "postgres" does not exist in laragon. how to fix this. thanks
![Screenshot (1201)](https://github.com/user-attachments/assets/1b017544-5f30-4f08-9987-d94757c4c09b)

**Reply by imunisasi** [reply]
same here
from laragon terminal
createuser --superuser postgres

credit from @KingGeneral and @bayukartiko comment below

**Comment by leokhoa** [comment]
@Pen-y-Fan: You are awesome. You know Laragon better than me :)
You can also add a tool calls Dbeaver in **Laragon menu > Tools > Quick Add > Dbeaver**.
It is also a great database client.

**Comment by DesarrollosDimpulsoX** [comment]
Thank You!!
your regards

**Comment by eokwukwe** [comment]
I have tried the above but can't connect to the DB. I get the error: `role "postgres" does not exist. 
In my case, I have to change the port for the Postgres in laragon because I already has postgres in my machine

**Comment by KingGeneral** [comment]
i think this will help for those who experience role postgres does not exist

`createuser --superuser postgres`

more : 
[psql-fatal-role-postgres-does-not-exist](https://stackoverflow.com/questions/15301826/psql-fatal-role-postgres-does-not-exist)

**Reply by bayukartiko** [reply]
for anyone who using custom port

`createuser --superuser postgres -p your_port_number`

**Comment by bloodykheeng** [comment]
---

I was having trouble installing **PostgreSQL** on Windows — the installation kept failing, and I couldn’t figure out why.

Later, I suspected something was already using the default PostgreSQL port, **5432**. To check this, I opened **Command Prompt as Administrator** (this is important for certain commands to work properly) and ran:

```bash
netstat -aon | findstr :5432
```

Output:

```
C:\Users\BK>netstat -aon | findstr :5432
  TCP    0.0.0.0:5432           0.0.0.0:0              LISTENING       6176
  TCP    [::]:5432              [::]:0                 LISTENING       6176
```
> This shows that **port 5432 is in use by a process with PID `6176`**, listening on all network interfaces.

To identify what was using that PID, I ran:

```bash
tasklist /FI "PID eq 6176"
```

Which returned:

```
Image Name                     PID         Session Name        Mem Usage
=========================     ========    ================    ============
postgres.exe                  6176        Console             25,000 K
```

So PostgreSQL was already running in the background and using port 5432.

To fix the issue, I did the following:

1. **Opened Command Prompt as Administrator** (⚠️ required to kill the process successfully).
2. Stopped the process using:

   ```bash
   taskkill /PID 6176 /F
   ```
3. Deleted the existing PostgreSQL data folder to clean up any broken or incomplete setup:

   ```
   C:\laragon\data\postgresql
   ```
4. Restarted **Laragon**, and everything worked perfectly — PostgreSQL started without issues.

---
